rust/core-lib/src/event_context.rs

```diff
@@ -225,6 +225,9 @@ impl<'a> EventContext<'a> {
             UpdateSpans { start, len, spans, rev } => self.with_editor(|ed, view, _, _| {
                 ed.update_spans(view, plugin, start, len, spans, rev)
             }),
+            UpdateAnnotations { start, len, type_id, spans } => self.with_editor(|ed, view, _, _| {
+                ed.update_annotations(view, plugin, start, len, type_id, spans)
```

Comment:
How do plugins update annotations? Is it right to have a separate RPC for this?

Reply:
There's no updating in place; to update the plugin just sends a new annotation.